src/shared/components/post/create-post.tsx

```diff
@@ -237,6 +249,7 @@ export class CreatePost extends Component<
           <div id="createPostForm" className="col-12 col-lg-6 offset-lg-3 mb-4">
             <h1 className="h4 mb-4">{I18NextService.i18n.t("create_post")}</h1>
             <PostForm
+              key={this.state.resetCounter}
```

Comment:
Using a key (outside of a collection) to prevent uneccessary rerenders. Clever.

Reply:
It still has to render even when the key stays the same. But the when the key changes it has to be recreated and resets its state. I added it so that the "Create Post" link in the top navigation completely resets the form.